packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -229,6 +228,7 @@ export function getIpfsCIDv1AndPath(ipfsUrl: string): {
   const cid = index !== -1 ? url.substring(0, index) : url;
   const path = index !== -1 ? url.substring(index) : undefined;
 
+  const { CID } = await import('multiformats');
```

Comment:
`multiformats` is an ESM-only package, and the core monorepo is CJS (even though it builds output for both CJS and ESM). 

With `NodeNext` enabled, CJS modules cannot import from ESM modules except through a dynamic import, because all static `import` statements are transpiled down to `require` statements, which do not support ESM modules.

The dynamic import in this line replaces the previous static import:
```ts
import { CID } from 'multiformats/cid'
```

Downstream, this line causes the following changes:

- `getIpfsCIDv1AndPath`, `getFormattedIpfsUrl` are now async functions (this is a breaking change for `assets-controllers`).
- The `--experimental-vm-modules` flag needs to be prepended to our build scripts that run `jest`, with accompanying updates to our yarn constraints.